internal/router/rpcservice/rpcservice.go

```diff
@@ -270,3 +274,34 @@ func (rs *RPCService) SetController(
 	ret.Transaction.Hash = txn.Hash().Hex()
 	return ret, nil
 }
+
+func (rs *RPCService) collectReadQueryMetric(ctx context.Context, statement string, config formatter.FormatConfig) {
+	value := ctx.Value(middlewares.ContextIPAddress)
+	ipAddress, ok := value.(string)
+	if ok && ipAddress != "" {
+		metric := &ReadQueryMetricData{
+			ipAddress:    ipAddress,
+			sqlStatement: statement,
+			extract:      config.Extract,
+			unwrap:       config.Unwrap,
+			output:       string(config.Output),
+		}
+		if err := telemetry.Collect(ctx, metric); err != nil {
+			log.Warn().Err(err).Msg("failed to collect metric")
+		}
+	}
+}
```

Comment:
Can we create a metric helper to avoid this [code duplication](https://github.com/tablelandnetwork/go-tableland/pull/385/files#diff-5588b773fe67092e7366bf6608cb74cae8d1475902b773f5c18a87e305b50dc2R273)?
This can also include the `ReadQueryMetricData` struct?

We can ignore this for a while since JSON-RPC will be gone in the medium term, but for a while both duplicates would exist. No strong opinions.

Reply:
Yeah. I'll think about this. Not sure where would be a good place to put it. After JSON-RPC deletion, it would stay in an awkward place, far from where it's used.